### PR TITLE
feat: onboard emersys destination 

### PR DESCRIPTION
**Pull request by shrouti1507**
## What are the changes introduced in this PR?

Identify call : https://dev.emarsys.com/docs/core-api-reference/f8ljhut3ac2i1-update-contacts
Group call : https://dev.emarsys.com/docs/core-api-reference/1m0m70hy3tuov-add-contacts-to-a-contact-list
Track call : https://dev.emarsys.com/docs/core-api-reference/fl0xx6rwfbwqb-trigger-an-external-event

## What is the related Linear task?

Resolves INT-1598

## Please explain the objectives of your changes below

Put down any required details on the broader aspect of your changes. If there are any dependent changes, **mandatorily** mention them here

### Any changes to existing capabilities/behaviour, mention the reason & what are the changes ?

N/A

### Any new dependencies introduced with this change?

N/A

### Any new generic utility introduced or modified. Please explain the changes.

N/A

### Any technical or performance related pointers to consider with the change?

N/A

@coderabbitai review

<hr>

### Developer checklist

- [x] My code follows the style guidelines of this project

- [x] **No breaking changes are being introduced.**

- [x] All related docs linked with the PR?

- [x] All changes manually tested?

- [x] Any documentation changes needed with this change?

- [x] Is the PR limited to 10 file changes?

- [x] Is the PR limited to one linear task?

- [x] Are relevant unit and component test-cases added?

### Reviewer checklist

- [ ] Is the type of change in the PR title appropriate as per the changes?

- [ ] Verified that there are no credentials or confidential data exposed with the changes.

### CODE REVIEW

**Comment by devops-github-rudderstack** [comment]
Test report for this run is available at: https://test-integrations-dev.s3.amazonaws.com/integrations-test-reports/rudder-transformer/3318/test-report.html

**Review by anantjain45823 — COMMENTED** [review]
LGTM but can we add respective doc like error code docs link at concerned places

**Comment by anantjain45823** [comment]
Have we verified the security hotspot is not dangerous and if yes can we close put some comment for the same ?

**Review by anantjain45823 — APPROVED** [review]

**Comment by shrouti1507** [comment]
> Have we verified the security hotspot is not dangerous and if yes can we close put some comment for the same ?

yes, but emarsys has already provided an auth function : https://dev.emarsys.com/docs/core-api-reference/b3c3a1eba8515-authentication

**Review by Gauravudia — APPROVED** [review]

File: src/cdk/v2/destinations/emarsys/procWorkflow.yaml [hunk 1]
```diff
@@ -0,0 +1,91 @@
...
+
+  - name: validateInputForTrack
+    description: Additional validation for Track events
+    condition: $.outputs.messageType === {{$.EventType.TRACK}}
+    template: |
+      $.assert(.message.event, "event could not be mapped to conversion rule. Aborting.")
+  - name: preparePayloadForIdentify
+    description: |
+      Builds identify payload.
+      ref: https://dev.emarsys.com/docs/core-api-reference/f8ljhut3ac2i1-update-contacts
```
Comment (koladilip) [inline src/cdk/v2/destinations/emarsys/procWorkflow.yaml:47]: looks like indentation issue with ref:

File: src/cdk/v2/destinations/emarsys/procWorkflow.yaml [hunk 2]
```diff
@@ -0,0 +1,91 @@
...
+    template: |
+      const properties = ^.message.properties;
+      const integrationObject = $.getIntegrationsObj(^.message, 'emarsys');
+      const emersysIdentifierId =  $.deduceCustomIdentifier(integrationObject, ^.destination.Config.emersysCustomIdentifier);
+      const payload = {
+          key_id: emersysIdentifierId,
+          external_id: $.deduceExternalIdValue(^.message,emersysIdentifierId,.destination.Config.fieldMapping),
+          trigger_id: integrationObject.trigger_id,
+          data: properties.data,
+          attachment: Array.isArray(properties.attachment)
```
Comment (koladilip) [inline src/cdk/v2/destinations/emarsys/procWorkflow.yaml:72]: you can use CommonUtils.toArray

File: src/cdk/v2/destinations/emarsys/procWorkflow.yaml (hunk 1 above)
Comment (shrouti1507) [reply src/cdk/v2/destinations/emarsys/procWorkflow.yaml]: done

File: src/cdk/v2/destinations/emarsys/procWorkflow.yaml (hunk 2 above)
Comment (shrouti1507) [reply src/cdk/v2/destinations/emarsys/procWorkflow.yaml]: done

**Review by anantjain45823 — APPROVED** [review]

**Review by Gauravudia — APPROVED** [review]

**Pull request merged**